### ISSUE TITLE
composite key filters

### ISSUE THREAD

**Issue by dillingham** [headlesslaravel/finders#14]
### Overview

[Filters](https://github.com/headlesslaravel/formations/wiki/Filters#general-filters) in this package are ways to accept parameters from the url.

These parameters are validated like POST requests to make sure ?status=invalid does not work for example

The parameter is then used as a [condition on the eloquent query](https://github.com/headlesslaravel/formations/blob/main/src/Formation.php#L236) a Formation is using to produce index results.

`->scope()` is a formation filter "type" documented [here](https://github.com/headlesslaravel/formations/wiki/Filters#scope-filters) which applies an [eloquent local scope](https://laravel.com/docs/8.x/eloquent#local-scopes)

A ArticleFormation will use an Article model.

A ArticleFormation will have a `Filter::make('status')->scope(),`

Article model will have a `scopeStatus` method and accept the query string value of `status`

---

### Goal

The goal of this issue is to allow multiple query string parameters to be used for 1 filter.

```
?year=1&make=2&model=3
```

A user interface would have select boxes for each of those values independently.

All 3 of those filters could have their own filters and model scope: ie `scopeYear`

```php
Filter::make('year')->scope('year')
Filter::make('make')->scope('make')
Filter::make('model')->scope('model')
```

But it would be better as 1 filter with 1 model scope to make 1 mysql condition seen here in headlesslaravel/filters#15 

> The goal of this issue is to allow multiple query string parameters to be used for 1 filter.

So basically I'd like to figure out a way to allow the parameters into the url but passed to 1 scope

---

### Outcome

"if the scope is the same..and all are present in the url.. pass query params to the same scope method 1 time in the order they are defined"

```php
<?php

class Product extends Model
{
    public function scopeVehicle($query, $year, $make, $model)
    {
        $query->where('year_id', $year);
        $query->where('make_id', $make);
        $query->where('model_id', $model);
    }
}
```

```php
Filter::scope('vehicle', [
  Filter::make('year')->rules('numeric')
  Filter::make('make')
  Filter::make('model')
]),
```
